be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/federation/FederationExpect.scala

```diff
@@ -0,0 +1,27 @@
+package ch.epfl.pop.model.network.method.message.data.federation
+
+import ch.epfl.pop.json.MessageDataProtocol.FederationExpectFormat
+import ch.epfl.pop.model.network.Parsable
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.model.network.method.message.data.{ActionType, MessageData, ObjectType}
+import ch.epfl.pop.model.objects.{Hash, PublicKey}
+import spray.json.*
+
+final case class FederationExpect(
+    laoId: Hash,
+    serverAddress: String,
+    other_organizer: PublicKey,
+    challenge: Message
+) extends MessageData {
+  require(serverAddress.matches("^(ws|wss):\\/\\/.*(:\\\\d{0,5})?\\/.*$"), s"This is an invalid server address")
```

Comment:
As mentionned during meeting, using "require" doesn't allow to handle errors gracefully. As you have this pattern on your jsonRPC, you could assume that if a message reaches here, it has the right format. Or you could try to handle it gracefully and return an error/default value in case of wrong format.